packages/web-react/src/types/textFieldBase.ts

```diff
@@ -1,23 +1,26 @@
-import { ChildrenProps, TextInputProps } from './shared';
+import { ChildrenProps, HelperTextProps, PasswordToggleAdornmentProp, TextInputProps, Validation } from './shared';
+import { LabelProps } from './label';
+import { MessageProps } from './message';
 import { TextAreaProps } from './textArea';
-import { TextFieldProps, TextFieldType } from './textField';
-import { PasswordToggleAdornmentProp } from './shared/adornments';
+import { TextFieldProps } from './textField';
 
 export interface TextFieldBaseMultiLineProps {
   /** Whether the input is TextArea. */
   isMultiline?: boolean;
 }
 
-export interface TextFieldBaseProps extends ChildrenProps, TextFieldProps, TextAreaProps, TextFieldBaseMultiLineProps {}
+export type TextFieldBaseCommonProps = ChildrenProps &
+  LabelProps &
+  MessageProps &
+  HelperTextProps &
+  TextInputProps &
+  Validation;
 
-export interface SpiritTextFieldBaseProps extends TextFieldBaseProps {}
+export type TextFieldBaseProps = ChildrenProps & TextFieldBaseMultiLineProps & (TextFieldProps | TextAreaProps);
```

Comment:
Why like this? `TextFielsBaseProp` should be common for both `TextField` and `TextArea` so why this type is using them? It seems weird to me.

Reply:
It's the exact opposite. As the component is designed now, it doesn't allow you to build those types any other way. Originally I just thought I could do it, but it requires a lot more intervention and therefore more time.
At the same time, there is a complication in that the TextFieldBase contains the PasswordToggle, which I think is wrong.

Reply:
But I renamed them and moved them instead to inputs.ts

Reply:
So, TextFieldBase can simply be an input or textarea element. Therefore, it now consists of both types that TextFieldBase requires.